src/rimage.c

```diff
@@ -86,11 +86,12 @@ int main(int argc, char *argv[])
 			usage(argv[0]);
 			break;
 		default:
+		 /* getopt's default error message is good enough */
 			break;
 		}
 	}
 
-	elf_argc = optind;
+	const int first_non_opt = optind;
```

Comment:
Everywhere in project we declare new variables at the begin of function, I guess we shouldn't make here exception and keep code style consistent. Of course, then const must be removed, but it's not so critical for local variables.

Reply:
This is a not just a code style question, this is an important new feature added by C99 to reduce the chance of memory corruption. Uninitialized variables is a cause of crashes and security issues.

I understand xt-xcc doesn't support C99 so it probably requires declarations at the start of a block (not function) but do we really need to keep user space tools stuck in the past too? C99 is 20 years old now.

Reply:
@marc-hb can you update this now, we can discuss this after v1.6. Thanks

Reply:
Done.